### PR TITLE
PanelState: Introduce a new separate redux panel state not keyed by panel.id 

### PR DESCRIPTION
**Pull request by torkelo**
A small step in a direction...  

Been thinking about PanelModel and DashboardModel and state, and how the current messiness of small parts in redux and small parts in mutable states is too messy and we need to do something about. Ran into more of these issues when working on panel instance state and when starting visualization suggestions this morning. And looking at the code for changing panel plugin (for normal panels but library panels as well) it's just sooooo complex and messy and full of timing issues. We have to do something, but very hard to do something without having to change the whole world. Struggled a lot in this PR to keep it small (had to rollback changes multiple times)  

Trying to solve many things here 
* [x] Get rid of the magic panel.id we set for edit panels, and panel.editSourceId. Causing lots of bugs old and existing (cannot create annotations from panel edit currently). We can get rid of this easily by using panel.key (and set this to unique id) instead of panel.id as the redux store key. 
* [x] Move panel store state out of dashboards to start to disentangle panels more from the dashboard (redux & DashboardModel), goal is to make Panel & PanelEditor less dependent on DashboardModel and dashboard redux state. But a long journey.  
* [x] Create a separate redux thunk to change a panel into a library panel, this is still a bit messy due to mutation but it's a bit better here. It has now a special thunk that does not call changePanelPlugin as that is not needed for this case. 
* [x] Update unit tests 

Issues 
* [x] Make changing panel plugin more clean. Still struggling a lot here. Mainly due to PanelModel still being mutable and panel.key changing in a mutable way causing issues. I actually tried to make changing panel type create a new PanelModel and update PanelEditor state etc to reflect the new PanelModel instance, but it requires a lot of changes to code that currently rely on panel model instance not changing. Not sure what next step is here, make PanelModel plugin change non mutable (new panel instance), or move something else. 

Update
- Resolve the panel.key mutation issue by passing a stateKey to DashboardPanel so it can use this when cleaning up state

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
Love this makes it so much easier to reason about. Let me know when you need help with manual testing. 🎉   ❤️

**Review by ashharrison90 — COMMENTED** [review]
minor nit: could we split future PRs like this more? e.g. moving files is easy to understand in a standalone PR, but becomes difficult when tangled with the other changes.

**Review by axelavargas — COMMENTED** [review]
This PR looks very promising @torkelo 🚀. Unfortunately, I find it hard to follow the changes; lack of context of how things work.

Is it possible to have a knowledge-sharing session in the frontend weekly or similar? 

I believe understanding the current architecture and where are we heading with this PR will help us to bring ideas or fix potential side effects in the future :)

**Review by hugohaggmark — APPROVED** [review]
Glorious work on this one! I've tested a bunch of different library panel scenarios, with and without alerts and annotations and I couldn't find any issues at all.
